homeassistant/components/sensor/folder.py

```diff
@@ -13,27 +14,49 @@
 
 from homeassistant.helpers.entity import Entity
 import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_NAME
 from homeassistant.components.sensor import PLATFORM_SCHEMA
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_FOLDER_PATHS = 'folder'
+CONF_FOLDER_PATH = 'folder'
 CONF_FILTER = 'filter'
+CONF_RECURSIVE = 'recursive'
 DEFAULT_FILTER = '*'
+DEFAULT_NAME = ''
+DEFAULT_RECURSIVE = False
+FILE = 'file'
+SIGNAL_FILE_ADDED = 'file_added'
+SIGNAL_FILE_DELETED = 'file_deleted'
+SIGNAL_FILE_MODIFIED = 'file_modified'
 
 SCAN_INTERVAL = timedelta(seconds=1)
```

Comment:
WHAT IS THIS

This is horrible. And should be removed. This is literally hammering peoples drives.

If you want INSTANT notification, you listen for filesystem events.